pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +405,62 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
+	errResp, _ := unwrapError(body)
+	if client.WaitOnJob { // check sync flag
+		unwrapMessage := &ErrorResponse{}
+		err := json.Unmarshal(body, unwrapMessage)
+		if err != nil {
+			return nil, err
+		}
+		ids, err := handleAsyncJob(unwrapMessage, client)
+
+		if err != nil {
+			return nil, fmt.Errorf("http response error: status (202), messages: %v", err.Error())
+		}
+		return ids, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+	}
+	return nil, fmt.Errorf("http response error: status (202), messages: %v", errResp)
+}
+
+//handleAsyncJob : extract the job_id and monitor it periodically until job completion or timed out
+func handleAsyncJob(resp *ErrorResponse, client *GroupMgmtClient) (string, error) {
```

Comment:
The purpose of this function seems a bit jumbled.  Maybe it would help if we move the job ID extraction back into processAsyncResponse at line 418 (since it is more like processing the response than handling the job) and rename this function to:

func waitForJobResult(jobId string, client *GroupMgmtClient) (string, error)

Reply:
Yes , it is due to the refactoring of common code . Anyways i will move jobiD extraction to processAsyncResponse.